bin/supply

```diff
@@ -16,26 +16,99 @@ BUILDPACK_DIR=$(dirname $BIN_DIR)
+  file ${BUILDPACK_DIR}/conjur-env/conjur-env
+
+  echo "[cyberark-conjur-buildpack]: copy the conjur-env binary from the buildpack dir to the dependencies/index directory"
+  mkdir -p ${DEPS_DIR}/${INDEX_DIR}/vendor/
+  cp -v ${BUILDPACK_DIR}/conjur-env/conjur-env ${DEPS_DIR}/${INDEX_DIR}/vendor/conjur-env
+popd
+
+
 pushd ${DEPS_DIR}/${INDEX_DIR}
   # We add the lib/0001_retrieve-secrets.sh script to profile.d so that it will
   # be run automatically to retrieve secrets when the app starts.
-  mkdir -p profile.d
+
+  echo "[cyberark-conjur-buildpack]: setup the dependencies/index/profile.d script so the conjur app starts automatically"
+  mkdir -p ./profile.d
   cp ${BUILDPACK_DIR}/lib/0001_retrieve-secrets.sh ./profile.d/0001_retrieve-secrets.sh
   sed "s/__BUILDPACK_INDEX__/$INDEX_DIR/g" ./profile.d/0001_retrieve-secrets.sh -i
 
+  # Its Complicated :D 
+  # Buildpacks are in a huge state of flux. As of Spring 2019, buildpacks use the 
+  # V2 apis, but the V3 is a fundamental change for the better, but it is a big change.
+  # Couple that with the fact that the Java Buildpack is *HUGE*.  So what is happening is
+  # the Java Buildpack is being deconstructed into multiple smaller java build pack 
+  # that are more focused on individual problems. 
+
+  # why does this matter? Because it is the final buildpacks whose job is to 
+  # look through all the ./profile.d/ directories of all the dependencies and execute scripts. 
+  # and the Java Buildpack v2 has not plans to implement this functionality.
+
+  # https://github.com/cloudfoundry/java-buildpack/issues/563#issuecomment-452437417
+
+  # :(
+
+  # so V2 supply buildpacks must *push* their scripts over into the /home/vcap/app/.profile.d/ 
+  # directory. This is breaking of the build pack contract to manipulate anything inside /home/vcap/app
+  # but at this point, you don't have much of a choice, all all this gets fixed in the V3 versions. 
+
+  # /shrug.  
+
+  # CONCERN: this following code, may only need to be executed *IF* the final build pack is java*
+  # This will need to be heavily tested with multiple final buildpacks. 
+  # or if the conjur-env runs twice is that a big deal? It will just 
+  # set the env twice.  not good, but not terrible.
+
+  echo "[cyberark-conjur-buildpack]:  - because of missing feature in Java Buildpack, also copy to /home/vcap/app/.profile.d/"
+  mkdir -p $BUILD_DIR/.profile.d/
+  cp -v ./profile.d/0001_retrieve-secrets.sh $BUILD_DIR/.profile.d/
+
   # conjur-env reads a secrets.yml file and uses it to retrieve secrets from
   # Conjur. We copy it to the dependency directory to make it accessible to the
   # /profile.d script. The /vendor subdirectory is just for convenience.
-  mkdir -p vendor
-  cp ${BUILDPACK_DIR}/vendor/conjur-env ./vendor/conjur-env
+  
+  # note, the 0001_retrieve-secrets.sh script referenced:
+  #   $DEPS_DIR/__BUILDPACK_INDEX__/vendor/conjur-env
+  # so why are we copying it to a different directory below? 
```

Comment:
That... is an excellent question. 😄 I dug into it a little bit, and it would appear this is a latent relic of earlier development.

I would suggest removing lines 101-111.

Reply:
Removed on next commit.